flocker/node/script.py

```diff
@@ -179,11 +132,47 @@ def get_service(self, reactor, options):
         """
         return AgentLoopService(
             reactor=reactor,
-            deployer=self.deployer_factory(hostname=options["hostname"]),
+            deployer=self.deployer_factory(options=options),
             host=options["destination-host"], port=options["destination-port"],
         )
 
 
+def volume_service_from_config(config):
+    pass
+
+def dataset_deployer_from_options(options):
+    """
+    Returns an IDeployer configured by options.
+    """
+    import yaml
+    config = yaml.safe_load(options['config-file'].getContents())
+    if config['backend'] == 'zfs':
+        # TODO The volume service should be integrated with the reactor - see
+        # main_for_service
+        volume_service = VolumeService(
+            config_path=flocker.volume.service.DEFAULT_CONFIG_PATH,
+            pool=config.get('zfs-pool', flocker.volume.service.FLOCKER_POOL),
+            # TODO thread through the reactor
+            reactor=reactor,
+        )
+
+        deployer_factory = partial(
+            P2PManifestationDeployer, volume_service=volume_service)
+    elif config['backend'] == 'loopback':
+        # Later, construction of this object can be moved into
+        # AgentServiceFactory.get_service where various options passed on
+        # the command line could alter what is created and how it is initialized.
+        loopback_pool = config.get('loopback-pool',
+                                   b"/var/lib/flocker/loopback")
+        api = LoopbackBlockDeviceAPI.from_path(loopback_pool)
+        deployer_factory = partial(
+            BlockDeviceDeployer,
+            block_device_api=api,
+        )
```

Comment:
```
if x == y:
   ...
elif x == z:
   ...
```

is an anti-pattern.  Do some dynamic dispatch (and we're halfway to pluggable backends).